packages/performance-tests/test/TransformerRegression.test.ts

```diff
@@ -102,17 +107,60 @@ async function runRegressionTests() {
   describe("Transformer Regression Tests", function () {
     testIModels.forEach(async (iModel) => {
       describe(`Transforms of ${iModel.name}`, async () => {
+        let sourceDb: BriefcaseDb;
+        let reportInfo: ReporterInfo
+
+        before(async () => {
+          Logger.logInfo(loggerCategory, `processing iModel '${iModel.name}' of size '${iModel.tShirtSize.toUpperCase()}'`);
+          sourceDb = await iModel.load();
+          const fedGuidSaturation = sourceDb.withStatement(
+            `
+            SELECT
+            CAST(SUM(hasGuid) as DOUBLE)/SUM(total) ratio 
+            FROM (
+              SELECT IIF(FederationGuid IS NOT NULL, 1, 0) hasGuid,
+              1 as total FROM bis.Element
+            )`,
+            (stmt) => {
+              assert(stmt.step() === DbResult.BE_SQLITE_ROW);;
+              return stmt.getValue(0).getDouble();
+            }
+          );
+          Logger.logInfo(loggerCategory, `Federation Guid Saturation '${fedGuidSaturation}'`);
+          const toGb = (bytes: number) => `${(bytes / 1024 **3).toFixed(2)}Gb`;
+          const sizeInGb = toGb(fs.statSync(sourceDb.pathName).size);
+          Logger.logInfo(loggerCategory, `loaded (${sizeInGb})'`);
+          reportInfo = {
+            "Id": iModel.iModelId,
+            "T-shirt size": iModel.tShirtSize,
+            "Gb size": sizeInGb,
+            "Branch Name": branchName,
+            "Federation Guid Saturation 0-1": fedGuidSaturation,
+          };
+        });
+
         testCasesMap.forEach(async (testCase, key) => {
+          before(async () => {
+            if(!sourceDb.isOpen){
+              sourceDb = await BriefcaseDb.open({
+                fileName: sourceDb.pathName,
+                readonly: true,
+              });
+            }
+          });
+
           it(key, async () => {
-            reporter = await testCase(iModel, reporter, branchName);
+            reporter = await testCase(sourceDb, reporter, reportInfo);
```

Comment:
is it worth instead doing like:

```ts
await testCase(sourceDb, {
  addReport: (smallReportSubset) => {
    reporter.addEntry( ...smallReportSubset, additionalInfo: reportInfo });
  }
);
```

This removes the two way relationship where the consumer must correctly send `reportInfo` into the `reporter`

Reply:
Not sure I fully understand what makes this a two way relationship? However, I needed `reportInfo` in order to get the branch name which is needed for the test name. If this still gives access to reportInfo to use in smallReportSubset, then this is a change I could make, but I guess I don't understand the benefit than?

Reply:
Can't we generate the test name here anyway? We know the iModel, test case, and branch here. This function could add a parameter for tests to add a subname if they want to add multiple entries

Reply:
still had trouble with this, so may need to discuss more. However used this to start modifying reporter system

Reply:
done, syntax slightly different, so please check if that is what you intended